### PR TITLE
Feature/nh story setup refactor

### PR DESCRIPTION
**Pull request by nickhartdev**
#### What's this PR do?
- Refactors StoryEdit to be a functional component
- Moves all state info and relevant methods to StoryEditView
- Refactors the fetch call to use getData instead of getRandomPrompt
#### Where should the reviewer start?
StorySetupView
#### How should this be manually tested?
Click new story, choose a prompt, and start your story! A prompt from the appropriate genre should render if the user chooses to use one.
#### Any background context?
I'll revisit this to take care of ticket #35. The original plan was to write methods to utilize the fact that App holds all prompts, but I ran out of time so the original fetch call is still there.
#### What are the relevant tickets?
None yet!
#### Screenshots (if appropriate)
cat

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "react-dom": "^16.13.1",
     "react-router-dom": "^5.2.0",
     "react-scripts": "3.4.3",
-    "react-tooltip": "^4.2.9"
+    "react-tooltip": "^4.2.10"
```
Comment (GreysonElkins) [inline package.json:16]: _**UPDATES**_

File: src/assets/testData/testData.js [hunk 2]
```diff
@@ -3,61 +3,62 @@ const testData = {
...
         'Make a little "birdhouse" in your soul',
         "Not to put to fine a point on it.",
         "Say I'm the only bee in your bonnet.",
       ],
+      contributors: [1, 2, 3, 1, 2],
       prompt: "sci-fi",
       created_at: "2020-10-10T22:37:51.103Z",
       updated_at: "2020-10-10T22:37:51.103Z",
-      isComplete: true,
+      is_complete: true,
```
Comment (GreysonElkins) [inline src/assets/testData/testData.js:44]: This was a goofy accident.

File: src/components/App/App.js [hunk 3]
```diff
@@ -6,44 +6,121 @@ import WelcomePageView from '../WelcomePageView/WelcomePageView'
...
+              const allAuthors = this.state.authors.concat(author)
+              story.contributors[i] = author
+              this.setState({ authors: allAuthors })
+            })
+        }
+      })
+    this.updateStoryData(story)
+  }
+
+  updateStoryData = (newStory) => {
```
Comment (GreysonElkins) [inline src/components/App/App.js:61]: @nickhartdev Did you comment these back in? These versions break the home page, right?

I have the fixes, waiting till everything's merged up to submit my next PR.

File: src/components/Bookshelf/Book.js [hunk 4]
```diff
@@ -18,9 +18,13 @@ const getBookSpine = () => {
   return bookSpines[random];
 };
 
-const Book = ({ story, onClick, popup }) => {
+const findLastWords = (story) => {
```
Comment (GreysonElkins) [inline src/components/Bookshelf/Book.js:21]: This is also breaking. I think I put a band-aid on it, but did not fix.

**Review by GreysonElkins — APPROVED** [review]
This PR actually has a lot of stuff I did that hasn't been merged yet, and likely has been adjusted... not sure what to do about that.

File: src/ApiHelper/ApiHelper.js [hunk 5]
```diff
@@ -1,21 +1,10 @@
...
-    const prompt = await response.json()
-    return prompt
-  }
-
-  static getData = () => {
-    return Promise.resolve(testData.stories)
+  static getData = (type, id) => {
+    return fetch(`${apiHead}/${type}/${id ? id : ''}`)
+      .then(response => response.json())
+      .catch(error => error.message)
```
Comment (carlymclift) [inline src/ApiHelper/ApiHelper.js:7]: Nice dynamic fetch method here!

File: src/components/LibraryView/LibraryView.js [hunk 6]
```diff
@@ -38,11 +27,12 @@ class LibraryView extends Component {
     return (
       <>
         <Bookshelf
-          stories={this.state.stories}
+          stories={this.props.stories}
+          authorUpdater={this.props.authorUpdater}
           onClick={this.selectStoryToRead}
-          popup={this.state.popup}
+          popup={false}
```
Comment (carlymclift) [inline src/components/LibraryView/LibraryView.js:33]: nice change here

**Review by carlymclift — APPROVED** [review]
Looks great! I approve! Thanks Nick

File: src/ApiHelper/ApiHelper.js (hunk 5 above)
Comment (Abdeboskey) [reply src/ApiHelper/ApiHelper.js:7]: Totally, that ternary is bangin.

File: src/components/StorySetup/StorySetup.js [hunk 7]
```diff
@@ -1,113 +1,67 @@
...
+        Who Are You?<br/>
+        <input
+          name='authorName'
+          type='text'
+          placeholder='create your pen name'
+          value={props.authorname}
+          onChange={props.handleChange}
+        />
+      </label>
+      }
```
Comment (Abdeboskey) [inline src/components/StorySetup/StorySetup.js:20]: I guess that once we have the login happening, we can go ahead and yet this conditional part.

File: src/components/StorySetupView/StorySetupView.js [hunk 8]
```diff
@@ -1,32 +1,86 @@
...
+    const value = target.type === "checkbox" ? target.checked : target.value;
+    const name = target.name;
+
+    this.setState({
+      [name]: value,
+    });
+  };
+
+  getPrompt = async genre => {
+    return await ApiHelper.getData("prompts", genre);
```
Comment (Abdeboskey) [inline src/components/StorySetupView/StorySetupView.js:32]: Do we need to put some error handling back in here?

**Review by Abdeboskey — APPROVED** [review]
Awesome work Nick, thank you!

**Pull request closed without merging**